components/script/dom/url.rs

```diff
@@ -24,19 +26,24 @@ pub struct URL {
 
     // https://url.spec.whatwg.org/#concept-urlutils-get-the-base
     base: Option<Url>,
+
+    // https://url.spec.whatwg.org/#dom-url-searchparams
+    query: JS<URLSearchParams>,
```

Comment:
Actually, this is a bit hairier if URLSearchParams is mutable through URL. It's also an iterator.

What happens if I mutate the original URL whilst iterating it?